brave/brave-core
# Add setting option for disabling tor component

fix https://github.com/brave/brave-browser/issues/6808

When tor is managed by group policy, this option is disabled. (only possible on Windows now)
![image](https://user-images.githubusercontent.com/6786187/69596130-e724f180-1044-11ea-9e48-befb6f7228b4.png)


## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
`yarn test brave_browser_tests --filter=BraveLocalStateBrowserTest*`
`yarn test brave_browser_tests --filter=TorDisabledPolicyBrowserTest*`
`yarn test brave_browser_tests --filter=TorEnabledPolicyBrowserTest*`
`yarn test brave_browser_tests --filter=NoTorPolicyBrowserTest*`

Manual test steps
1. Launch browser with clean profile
2. Load brave://settings/extensions and check `Tor` is on by default
3. Turn it off and Tor menu is not visible in the app menu
4. Turn it on and Tor menu is visible in the app menu
5. Open tor window
6. Turn it off and tor window is still alive
7. Close opened tor window and check Tor utility process is not visible from task manager.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
